src/filters/path.rs

```diff
@@ -485,10 +485,19 @@ fn path_and_query(route: &Route) -> PathAndQuery {
 /// that the shorter path filter does not match the longer paths.
 #[macro_export]
 macro_rules! path {
-    (@start $first:tt $(/ $tail:tt)*) => ({
-        let __p = path!(@segment $first);
+        ($($pieces:tt)*) => ({
+            use $crate::internal_path;
+            $crate::internal_path!(@start $($pieces)*)
+        });
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! internal_path {
```

Comment:
To follow the pattern, let's add a comment that it's not a public API, and prefix with double underscores.